tests/unit/mathematics/linalg/Linalg_dot_unittest.cc

```diff
@@ -0,0 +1,60 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+#ifdef HAVE_VIENNACL
+#include <shogun/mathematics/linalg/LinalgBackendViennaCL.h>
+#endif
+
+#include <shogun/lib/SGVector.h>
+#include <gtest/gtest.h>
+
+using namespace shogun;
+TEST(LinalgDot, linalg_cpu_backend_dot)
+{
+	const index_t size = 10;
+	SGVector<int32_t> a(size), b(size);
+	a.set_const(1);
+	b.range_fill(0);
```

Comment:
this method should be remove of SGVector and be part of linalg.
Put that on your todo list :)